engine/engine.go

```diff
@@ -89,71 +83,56 @@ func Start(ctx context.Context, startOpts *Config, fn StartCallback) error {
 	}
 	solveOpts := bkclient.SolveOpt{
 		Session: []session.Attachable{
-			secretsprovider.NewSecretProvider(secretStore),
 			socketProviders,
 			authprovider.NewDockerAuthProvider(os.Stderr),
 		},
 	}
-	if startOpts.LocalDirs == nil {
-		startOpts.LocalDirs = map[string]string{}
-	}
-	// make workdir ID unique by absolute path to prevent concurrent runs from
-	// interfering with each other
-	workdirID := fmt.Sprintf("%s_%s", workdirID, startOpts.Workdir)
-	startOpts.LocalDirs[workdirID] = startOpts.Workdir
-	solveOpts.LocalDirs = startOpts.LocalDirs
 
 	ch := make(chan *bkclient.SolveStatus)
+
+	session := core.NewSession(c, solveOpts, ch)
```

Comment:
Ok admittedly it's kind of confusing seeing "NewSession" here when we expressly _don't_ open a Buildkit session at this point anymore. :sweat_smile: